neal/sampler.py

```diff
@@ -302,30 +302,37 @@ def _default_ising_beta_range(h, J):
 
     Assume each variable in J is also in h.
 
-    We shoot for a final normalized beta of 10. We use the minimum bias to
-    give a lower bound on the minimum energy gap, such at the final sweeps
-    we are highly likely to settle into the current valley.
+    We use the minimum bias to give a lower bound on the minimum energy gap, such at the
+    final sweeps we are highly likely to settle into the current valley.
     """
-
-    bias_sum = 0.0
-    sigmas = {}
-    min_bias = float("inf")
-    for v, b in iteritems(h):
-        bias_sum += b**2
-        abs_b = abs(b)
-        sigmas[v] = abs_b
-        min_bias = min(abs_b or float("inf"), min_bias)
-    for (u, v), b in iteritems(J):
-        bias_sum += b**2
-        abs_b = abs(b)
-        sigmas[u] += abs_b
-        sigmas[v] += abs_b
-        min_bias = min(abs_b or float("inf"), min_bias)
-
-    if len(sigmas) > 0:
-        beta_range = [1.0/np.sqrt(bias_sum), 5.0/min_bias]
-    else:
-        # completely empty problem, so beta_range doesn't matter
-        beta_range = [0.1, 1.0]
-
-    return beta_range
+    # Get nonzero, absolute biases
+    abs_h = [abs(hh) for hh in h.values() if hh != 0]
+    abs_J = [abs(jj) for jj in J.values() if jj != 0]
+    abs_biases = abs_h + abs_J
+
+    if not abs_biases:
+        return [0.1, 1.0]
+
+    # Rough approximation of min change in energy
+    min_delta_energy = min(abs_biases)
+   
+    # Combine absolute biases by variable
+    abs_bias_dict = {k: abs(v) for k, v in h.items()}
+    for (k1, k2), v in J.items():
+        abs_bias_dict[k1] += abs(v)
+        abs_bias_dict[k2] += abs(v)
+
+    # Find max change in energy
+    max_delta_energy = sum(abs_biases) - min(abs_bias_dict.values())
```

Comment:
The `.items()` and `.values()` calls should probably be changed to `iteritems()` and `itervalues()`

Reply:
I think we're  starting to get less picky about python 2.7 performance, and in my opinion this improves readability. I'd like to start phasing out `six` where it is not necessary.